modin/pandas/test/test_io.py

```diff
@@ -2152,16 +2229,45 @@ def test_read_sql(self, tmp_path, make_sql_connection, read_sql_engine):
             con=sqlalchemy_connection,
         )
 
+        old_sql_engine = ReadSqlEngine.get()
         ReadSqlEngine.put(read_sql_engine)
         if ReadSqlEngine.get() == "Connectorx":
             modin_df = pd.read_sql(sql=query, con=conn)
         else:
             modin_df = pd.read_sql(
                 sql=query, con=ModinDatabaseConnection("sqlalchemy", conn)
             )
+        ReadSqlEngine.put(old_sql_engine)
```

Comment:
this should be done in `try..finally` to be absolutely sure we return the value back... not critical in tests I suppose, so mostly a note for the future patterns